### ISSUE TITLE
Revisit Cards Visual Design

### ISSUE THREAD

**Issue by joseegm** [SAP/techne#678]
### Goals
1. Create cards that can hold images as well as textual information in different variations
2. Refresh the visual style to be modern.
3. Update the hover effect to be less "jumpy" and more visible.
4. Update the Static and Disabled state.

### Prior art
v1.5.x - https://techne.yaas.io/patterns/Cards.html

Normal cards
<img width="989" alt="screen shot 2017-05-15 at 22 07 04" src="https://cloud.githubusercontent.com/assets/22662903/26076982/e909021a-39ba-11e7-9f60-6b853a0ef029.png">

Stacked cards
<img width="979" alt="screen shot 2017-05-15 at 22 08 29" src="https://cloud.githubusercontent.com/assets/22662903/26077043/153064dc-39bb-11e7-9681-babab2524faa.png">

Grouped Cards
<img width="978" alt="screen shot 2017-05-15 at 22 09 10" src="https://cloud.githubusercontent.com/assets/22662903/26077094/31f2441e-39bb-11e7-9387-ed9c34420bb1.png">

**Comment by joseegm** [comment]
Decided from #679 to removed the **Grouped Cards**

**Comment by joseegm** [comment]
@LeoT7508 Looking great!
Can you specify the margins inside the cards and also the minimun/default sizes?
Also describe the hover effect. Loos like a inner light blue shadow and small shadow outside.
That will help with the implementation.

**Comment by LeoT7508** [comment]
@joseegm 
![techne_2 0_cards](https://cloud.githubusercontent.com/assets/28761982/26415172/33c5096c-4077-11e7-8e42-731abd392442.jpg)

**Comment by joseegm** [comment]
@LeoT7508 
Let's change the spacing of 28px between "Secondary Value" and "Label" to 30px, to keep the base 10 for spacing. Easy to use, calculate and remember.

**Comment by joseegm** [comment]
### Implementation Details

Suggested CSS Structure
```css
.tn-cards {} /* main container */

    .tn-card {} /* main element */

        /* modifiers */

        .tn-static {} /* static status, no clickable, no hover effect */

        .tn-disabled, tn-card[disabled] {} /* light gray backround and font colors */

        .tn-card-with-image {} /* cards with place for image */

            .tn-card-vertical {} /* position the image element on top */
        
            .tn-card-big {} /* increase the size of the card to display the image bigger */

``` 
**Note**: `.tn-static` and `tn-disabled` dont have "card" to be able to keep these modifier consistent with other library elements.

Suggested HTML
```html
<ul class="tn-cards">

    <li class="tn-card">
        <div class="tn-card-content"></li>
    </li>

    <li class="tn-card tn-static"></li>

    <li class="tn-card tn-disabled"></li>

    <li class="tn-card tn-card-with-image">
        <div class="tn-card-image"></div>
        <div class="tn-card-content"></div>
    </li>

    <li class="tn-card tn-card-with-image tn-card-vertical"></li>

</ul>
```

**Comment by joseegm** [comment]
@denisewildner @saadmhybris @like-me1 @xak 
Take a look and see if we are missing something.

**Comment by denisewildner** [comment]
The values on the cards doesn't reflect worst case scenarios, such as having more secondary value aka description text, or a Button or two status information on the card (see current subscription cards). Should cards adapt to the content and grow with it in y-direction, or does the content need to be abbreviated? And if the card grows with the content, what happens to the image? Will it still be square format then? 
I also wonder why we don't have any margin on the right. Would text be cut of at the end of the card then?
Thanks for clarifying :)
![screen shot 2017-05-26 at 12 07 54](https://cloud.githubusercontent.com/assets/7449143/26490571/2acc5540-420c-11e7-8afc-357765960c2a.png)

**Comment by xak** [comment]
I started specing the Cards yesterday. See https://github.com/SAP/techne/wiki/Component-Specs:-Collections#cards

**Comment by joseegm** [comment]
This is not to be close... yet, wrong button pressed, most probably :D

There have been a lot of different uses of the cards, like these two

![screen shot 2017-05-26 at 13 44 51](https://cloud.githubusercontent.com/assets/22662903/26493575/abd48690-421a-11e7-838f-141de69e148b.png)

* on **Client Authorization** they have a lot of information and 2 actions at the bottom. On this case the new cards should provide space for the info and the actions, but not the 2 columns layout inside the cards. Or should we?

* on **Required Scopes** that can be replaced by a simple list, it is only the name and the "delete" action.

**Comment by joseegm** [comment]
@LeoT7508 
Think wee need one card with actions on the footer, the place where the label is. 
* One "primary" action and the other "secondary", might be a button and a link respectively.

**Comment by denisewildner** [comment]
![screen shot 2017-05-26 at 14 54 33](https://cloud.githubusercontent.com/assets/7449143/26495268/450f84a6-4223-11e7-8dc2-f7ac8e5850be.png)

**Comment by denisewildner** [comment]
![screen shot 2017-05-26 at 14 55 44](https://cloud.githubusercontent.com/assets/7449143/26495303/6c19764c-4223-11e7-9f3d-4eeb1c56970c.png)

**Comment by denisewildner** [comment]
![screen shot 2017-05-26 at 14 56 30](https://cloud.githubusercontent.com/assets/7449143/26495330/861e8a0a-4223-11e7-848c-789a0c6ddf33.png)

**Comment by denisewildner** [comment]
![screen shot 2017-05-26 at 14 57 29](https://cloud.githubusercontent.com/assets/7449143/26495375/abc24bb6-4223-11e7-84be-af7d52f738b5.png)

**Comment by denisewildner** [comment]
![screen shot 2017-05-26 at 15 00 38](https://cloud.githubusercontent.com/assets/7449143/26495487/1a0a2c06-4224-11e7-8c1f-1ad05e04095f.png)

**Comment by denisewildner** [comment]
![screen shot 2017-05-26 at 15 01 24](https://cloud.githubusercontent.com/assets/7449143/26495532/3aaa001c-4224-11e7-8a78-febd8a3d538c.png)

**Comment by denisewildner** [comment]
I have attached the different card scenarios we are currently having in the Builder. Let me know if you have any questions. :)

**Comment by xak** [comment]
My thought is that cards should have generic layout and specific implementations. We can't anticipate all uses so a dev could use the card only with the `image`, `content`, and `header` sections, for example, but then customize the content with other components. We can show specific examples like packages and products even if those are not specific types of cards, meaning we may not need `.tn-card--product`. It's more abstract.

Also, instead of having an `inactive` state, I recommend using a `--button` modifier when the card is a button. For accessibility, any card that triggers an action should include a role, like `<div class="tn-card" role="button">` or use an actual button like `<button class="tn-card">... card content...</button>`. This means only button cards would have a hover state.

**Comment by joseegm** [comment]
The case when the card is used to get to the details page of the object. But have some actions as well, will not be semantically correct to put it in a 'button' with other buttons or links inside. So, we still need a inactive case that works with any element, li, div, etc. Hover states are also the same.

**Comment by xak** [comment]
@joseegm Yes, any element can be used as a button, the CSS supports that. 

We can add an `inactive` state but that would just be the default state. The `--button` modification would only be concerned with high-level stuff like borders, outlines, backgrounds for various states and not affect additional links inside the "button" container.  

Cards are unique in that they mix layout and content so we should be cognizant of that separation.

**Comment by LeoT7508** [comment]
@joseegm @denisewildner 
So after a lot of thought, feedback from Jeanne, Zack, and more than a few variations heres what I came up with.

I created a stick set of rules for cards and they are as follows:

Cards will always only have:
- Primary value (wrap if necessary) 
- Secondary value (one line with “…” never wraps for spacing consistency)
- Status Label (one line, if there are multiple status’ they are separated by a 1px vertical rule,                 wrapping if necessary)
- Options (If necessary) 3 dots icon will hold all options

Also to answer a few questions Jose had:
- I updated the spacing from the Secondary Value to the Label to 30px. 
- The maximum width should be 500px, the minimum should be 370, and the padding between cards should be 20px. This should never change, I experimented with different values and this works best visually in all scenarios. 

I also feel strongly that images should always sit on top of information and never sit on the left, this will help with spacing and grid consistencies. But we can talk more about that in this string. 
![techne_2 0_cards_commoncards](https://cloud.githubusercontent.com/assets/28761982/26697706/950d99c0-46d8-11e7-936c-2c60bc13474e.jpg)
![techne_2 0_cards_with supporting image](https://cloud.githubusercontent.com/assets/28761982/26697707/9519d79e-46d8-11e7-9337-44ccb543ace5.jpg)

**Comment by jeannewalters** [comment]
I want to share some concerns about the image on top. In many cases, the image is not the priority and putting it in the top center is making it the priority. If it's a product, sure, it makes sense, but in most cases that I know of, the card does not represent a product - especially with CaaS not using it anymore. It seems that if teams follow the guidelines you have above about one line of text per line, the card heights will not be the issues it is currently is in some places in Builder. I also feel like having a mix of sections with cards on one page when some are centered and others are left aligned will be strange from a scannable perspective.  @joseegm I would like to strongly suggest that you identify all the current use cases (Builder, Smart Edit, Personalization) for cards. In addition, when designing each component, it is designed within the context of a sample page - not just a component. You are potentially losing context of what happens when mixing components, subheads, actions (add link or button), etc.  I just learned how to follow this GitHub project so you should be seeing a lot more of me on here the future.

**Comment by LeoT7508** [comment]
@jeannewalters @joseegm 

"but in most cases that I know of, the card does not represent a product - especially with CaaS not using it anymore. "

If this is the case then I would suggest eliminating the images from these cards. It seems that adding an image where its mostly useless only adds unnecessary information, takes up space, and clutters up the UI.

**Comment by denisewildner** [comment]
I agree @LeoT7508 with that in some cases the image might not be necessary, but we would definitely need them for members. Referring to @jeannewalters comment: The image in the middle looks a bit lost to me. In this size it hasn't any more value, it just has nice white space around ;)

What happens if there's only 1 Action button on the card, e.g. "UPDATE"? Show the three dots still? Or just 1 icon instead? As tests and experience proofs, it's often better to have a text instead of an icon for this kind of use cases. It doesn't need to be as big and yellow as it is now, but I would love to have a place for it. And in this case it's a pretty important action, too. Thanks

**Comment by joseegm** [comment]
To bring all back to the right context, since we are deviating a little bit from the main topic here.

The cards with Images we already have a v1.5.x. At this point we are porting the components we have on v1.5.6 as per this issue goal on https://github.com/SAP/techne/issues/678#issue-228829317. Let's stick to the goal otherwise this will go on forever without any useful outcome.

Current usages of cards with images: https://github.com/SAP/techne/issues/678#issuecomment-304275537, https://github.com/SAP/techne/issues/678#issuecomment-304275900, https://github.com/SAP/techne/issues/678#issuecomment-304276899

From the afore mentioned uses cases most of the images are on the left.

So we have to provide that as we do right now on v1.5.x
https://techne.yaas.io/patterns/Images-Avatars.html

Finally we already have the design done at: https://github.com/SAP/techne/issues/678#issuecomment-303784339

We will including the cards with images since is a current pattern and not having those will put the new version in disadvantage against 2.0. And nobody wants that, right @jeannewalters?

Like the rule of having only one column on the card only. Introduced on https://github.com/SAP/techne/issues/678#issuecomment-305599938

@denisewildner Question on https://github.com/SAP/techne/issues/678#issuecomment-305711217
This is going to be a separate issue, so we can focus on that in the right context, already created #701

The point made on https://github.com/SAP/techne/issues/678#issuecomment-305711217 about the cards with the circle image at the top, holds all validity. 
That type of card doesn't mage much sense as it is right now. Opposite to the ones with the circle on the left, that btw we already provide at: https://techne.yaas.io/patterns/Images-Avatars.html

**Comment by joseegm** [comment]
We have enough elements now on to decide which uses case we are implementing on Techne 2.0

## Card to be implemented

### 1. Common cards
![950d99c0-46d8-11e7-936c-2c60bc13474e](https://cloud.githubusercontent.com/assets/22662903/26717490/e0c66b58-477d-11e7-9693-95c47c6c3d4f.jpg)

### 2. Cards with images
![33c5096c-4077-11e7-8e42-731abd392442](https://cloud.githubusercontent.com/assets/22662903/26717599/47d1a664-477e-11e7-8fac-9ea3610f109a.jpg)
![9519d79e-46d8-11e7-9337-44ccb543ace5](https://cloud.githubusercontent.com/assets/22662903/26717664/7da839d8-477e-11e7-8ed1-f7dbb53dca06.jpg)

**Comment by LeoT7508** [comment]
@denisewildner 

I can't really agree with:

"The image in the middle looks a bit lost to me. In this size it hasn't any more value, it just has nice white space around ;)" 

Since we already use that size on the live site here:
![screen shot 2017-06-02 at 9 49 47 am](https://cloud.githubusercontent.com/assets/28761982/26731271/1f16fe5e-4779-11e7-9da0-011b20272ea5.png)

I think we're missing the point of what I was trying to accomplish here. I'm trying to eliminate instances like this:
![screen shot 2017-06-02 at 9 54 32 am](https://cloud.githubusercontent.com/assets/28761982/26731382/8e9ad502-4779-11e7-88fc-134e99561fae.png)

Where we have a ton of wasted space on the right side of these cards, or where the sentences get too long and stretch far to the right. 

The point of all these exercises is to create more consistency, creating rules for cards will help with inconsistent vertical and horizontal spacing in grids of cards. This is also the reason I added the 3 dots, to keep the vertical space of every card the same in any and all instances. If we have a button we run the risk of this:
![2acc5540-420c-11e7-8afc-357765960c2a](https://cloud.githubusercontent.com/assets/28761982/26731595/32b6d6a4-477a-11e7-917f-45e84b2936e1.png)

Where nothing lines up and card heights are out of control. Jeanne has mentioned scanability a lot lately, and having information staggered in cards because of inconsistency reduces scanability. These are all reasons I've designed these cards with these rules. 

One thing I would stress to everyone to keep in mind during these redesigns is what we already have doesn't work, we have a chance to fix the issues now, so lets not redesign something but want to make it the same as what we already have. 

@joseegm 

Please use this grid for Cards with supporting images. Per Jeanne's feedback about images being too prominent in some instances sticking with this layout will keep things most consistent. I want to eliminate as many variations as possible. 
![techne_2 0_cards_with supporting image](https://cloud.githubusercontent.com/assets/28761982/26731969/7ce7267e-477b-11e7-9127-9332c2d75c90.jpg)

**Comment by denisewildner** [comment]
@LeoT7508 I was only talking about this: ;)
![screen shot 2017-06-06 at 09 24 07](https://cloud.githubusercontent.com/assets/7449143/26818209/10ea1fea-4a9a-11e7-920c-7338240c8c0b.png)

**Comment by joseegm** [comment]
@LeoT7508 
Denise was only referring to the one in the comment before. That one we don't need.

We should keep in mind that it is okay simplifying things but we still have to provide for the current cases that makes sense to keep. The cards with images on the left are used in Builder currently.

The final types of cards are still the same as per 
https://github.com/SAP/techne/issues/678#issuecomment-305724490

**Comment by joseegm** [comment]
I think we are in a good place to start implementing.

@xak @saadmhybris @like-me1 Take a look, and you can start implementing it.
The implementation details is at #692 
Please not the reduced scope at https://github.com/SAP/techne/issues/692#issuecomment-305160067

**Comment by xak** [comment]
@joseegm There are 34 comments here and a lot of information. It would be very helpful if you gather the final approved designs and make them available for development. 

Also, I would like to see a design (or at least a wireframe) final page implementation with the final Cards. We won't need this for every component but we have nothing now except the current site but our new set here is more robust than the existing Cards. For example, are we keeping Product Thumbnail as a separate component or is it a new flavor of card? 

If you launch the site you can see some rudimentary examples at `buttons.html` and `tables.html`. We need to reconcile the patterns used for the component pages. Help me understand how everything should be presented and annotated. If you have any thoughts on the vocabulary around what each of these card types will be called that would be very useful.

We also need to verify the [specs on the Wiki](https://github.com/SAP/techne/wiki/Component-Specs:-Collections#cards) and keep in mind the data structure and how these will be output with constructor methods ...
```
{{ button(label="Update", modifier="secondary") }} 
```

All of this needs to be settled before we begin implementation so any help would be appreciated. Hopefully, we an begin this process much earlier as we start reviewing other components.

**Comment by joseegm** [comment]
@xak 
That was resumed on https://github.com/SAP/techne/issues/678#issuecomment-305724490 one week ago. 
And  yesterday again on https://github.com/SAP/techne/issues/678#issuecomment-306723044

The design of the cards have name themselves. If you think the names are not suitable please feel free to suggest others.

The specs in the wiki seems to be aligned with all discussed here. Otherwise please change it and let us know.

The components page details are at #694

What other information you need now?

**Comment by xak** [comment]
https://app.box.com/folder/29116880878

**Comment by joseegm** [comment]
@xak 
The first link to invision is not working.

Please if you want to use invision make sure that all links are working, otherwise please add the latest designs here where is easy accesible.
